pkg/object/ingresscontroller/ingresscontroller.go

```diff
@@ -0,0 +1,239 @@
+/*
+ * Copyright (c) 2017, MegaEase
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ingresscontroller
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/megaease/easegress/pkg/logger"
+	"github.com/megaease/easegress/pkg/object/httpserver"
+	"github.com/megaease/easegress/pkg/object/trafficcontroller"
+	"github.com/megaease/easegress/pkg/supervisor"
+)
+
+const (
+	Category = supervisor.CategoryBusinessController
+	Kind     = "IngressController"
+
+	defaultIngressClass          = "easegress"
+	defaultIngressControllerName = "megaease.com/ingress-controller"
+	k8sIngressClassAnnotation    = "kubernetes.io/ingress.class"
+)
+
+type (
+	// IngressController implements a k8s ingress controller
```

Comment:
`k8s` should be `K8s`

## Reference
1. https://kubernetes.io/docs/concepts/overview/what-is-kubernetes/